### PR TITLE
docs: sprint retrospective improvements (2026-04-28)

### PR DESCRIPTION
**Pull request by ms2sato**
## Summary

Apply 9 process improvements derived from Sprint 2026-04-28 retrospective.

### Cross-cutting incidents this addresses

- **Concerns Surfacing predictive failure (Pattern A)** — PR #710 Bootstrap gap and PR #715 preflight-check omission were both rationalized away by the Orchestrator and caught only because the owner asked. Q10 + Concerns Surfacing Discipline convert "owner happened to ask" into "Orchestrator surfaces unconditionally".
- **Cross-runtime spawn blind spot** — PR #716 first push failed `coverage-check` because `test-coverage-check.yml` lacked `setup-bun` even though that workflow now invokes a `node` script that spawns `bun`. The Orchestrator misdiagnosed the failure as a logic bug; the agent traced the spawn ENOENT chain to the true cause. Layer-Boundary Crossing Checklist + the reverse case in CI Self-Diagnosis close both sides.
- **Boundary value blind spot** — PR #711 `splitContentIntoChunks` passed unit tests but split UTF-16 surrogate pairs; CodeRabbit's GitHub bot caught it. Delegation prompt now lists surrogate-pair handling as a mandatory boundary case.
- **Bug report repro velocity** — Issue #708's reporter listed 4 ranked hypotheses; the agent confirmed Hypothesis 3 in TDD without rediscovery from scratch. New bug-report template codifies the pattern.

### Files changed

| File | Change |
|---|---|
| `.claude/skills/orchestrator/acceptance-check.js` | + Q10 "Concerns Surfacing & Owner Communication Gate" (5-step walk, HOLD on skip) |
| `.claude/skills/orchestrator/__tests__/acceptance-check.test.js` | + Q10 tests (count, key order, focus content) |
| `.claude/skills/orchestrator/core-responsibilities.md` | + "Concerns Surfacing Discipline" subsection under Acceptance Check |
| `.claude/rules/workflow.md` | + reverse case in CI Self-Diagnosis (env differences); + CLI/bot independent review depth note |
| `.claude/rules/pre-pr-completeness.md` | + Question 6 Layer-Boundary Crossing Checklist; + Question 3.5 fs.watch inode-binding sibling |
| `.claude/skills/orchestrator/delegation-prompt.js` | + Boundary Values section (surrogate / combining / empty); + 30% Checkpoint Reporting recommendation |
| `.github/ISSUE_TEMPLATE/bug-report.yml` | new template with Hypotheses (rank-ordered if known) field |

### Follow-up Issues (filed separately)

- #717 Letterlike Symbols / math notation handling in language lint regex
- #718 Source-code comment language lint (.ts / .js)
- #719 commit-msg / pre-commit hook for language check
- #720 Rename `coverage-check` workflow to reflect actual scope (preflight)

## Test plan

- [x] `bun run check:lang` exit 0 (98 files scanned, 0 violations)
- [x] `bun run typecheck` exit 0 across all packages
- [x] `bun run test` exit 0 — 4195 pass / 3 skip / 0 fail (client 1361 / shared 309 / integration 29 / server 2463 / scripts 33)
- [x] `bun test ./.claude/skills/orchestrator/__tests__/acceptance-check.test.js` — 65 pass / 0 fail
- [ ] CI rollup green via `gh pr view <PR> --json statusCheckRollup`
- [ ] CodeRabbit 3 layers clean

## Note on CodeRabbit

If local CodeRabbit CLI is rate-limited during this sprint, relying on GitHub-side CodeRabbit bot review (per `workflow.md` rate-limit fallback).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/rules/pre-pr-completeness.md [hunk 1]
```diff
@@ -27,13 +28,25 @@ Before opening a PR that introduces a **new skill, script, rule, file type, or c
    - Prefer concrete examples or file paths over abstract verbs ("check file X" beats "verify appropriately").
    - Remove prediction-framed statements (e.g., "X will fade") — rules describe what to do, not what the ecosystem will become.
 
+6. **Layer-Boundary Crossing Checklist — for PRs that introduce a cross-runtime spawn:**
+
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply .claude/rules/pre-pr-completeness.md:32]: Addressed in commit c10bedc — updated header to "6-question" / "six questions" wording.

File: .claude/rules/workflow.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ Before completing any code changes, always verify:
    ```
    Before merge, confirm the GitHub bot review is APPROVED. If `CHANGES_REQUESTED`, follow the resolution flow in `core-responsibilities.md` §6 (CodeRabbit "CHANGES_REQUESTED" resolution). (Sprint 2026-04-25 — verified in PRs #687 / #688 / #691 / #694.)
 
+   **CLI and GitHub bot have independent review depth.** The local CodeRabbit CLI and the GitHub-side CodeRabbit bot run on separate tokens and likely separate prompts; their findings do not always overlap. A clean local CLI pass does not guarantee a clean bot review, and vice versa. Treat the bot as an additional layer of detection rather than a fallback. **Do not merge while waiting for the bot review** — the rate-limit fallback note above defers merging only until the bot's APPROVED state is observable. (Sprint 2026-04-28 PR #711 — local CLI flagged 1 minor finding; the GitHub bot subsequently caught two Major findings the CLI had missed: an `await` propagation gap and a UTF-16 surrogate pair split. Both required follow-up commits before merge.)
+
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply .claude/rules/workflow.md:31]: Addressed in commit c10bedc — clarified that empty `reviewDecision` means "the bot has not yet reviewed; do not merge yet". Merge requires either observable APPROVED or owner-side dismissal of CHANGES_REQUESTED.

File: .claude/skills/orchestrator/acceptance-check.js [hunk 3]
```diff
@@ -433,6 +433,23 @@ function getQuestions(hasAcceptanceCriteria, { integrationTestMissing = false, l
...
+        '  5. If any concern is not surfaced → verdict is HOLD until the report is sent and owner has acknowledged.',
+        'PASS-with-notes is not a substitute for HOLD. The discipline is mechanical because LLM self-review is structurally weak at "what else should I be worried about that I rationalized away". (Lesson: Sprint 2026-04-28 PR #710 — Bootstrap gap was rationalized away; owner caught it. PR #715 — preflight integration was rationalized away; owner caught it. Both are structurally identical.)',
+      ].join('\n  '),
+      insufficient: '"Concerns are minor, proceeding with PASS" (silent skip — exactly the failure mode this question prevents)',
+      sufficient: '"Entry points: register_delegation_template MCP tool. Bootstrap: undefined who/when/where. Concerns: (a) dead-code on merge, (b) boilerplate likely duplicated with MemoService, (c) cross-session isolation untested. All three surfaced to owner via memo + dedicated message before this verdict; owner replied to defer to next sprint. Verdict: HOLD pending Bootstrap design."',
+    },
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply .claude/skills/orchestrator/acceptance-check.js:452]: Addressed in commit c10bedc — updated header docstring (Line 7) and usage error message (Line 39) from Q1-Q9 to Q1-Q11.

File: .claude/skills/orchestrator/core-responsibilities.md [hunk 4]
```diff
@@ -138,8 +138,27 @@
...
-  - **Preflight check** (mechanical): `node .claude/skills/orchestrator/preflight-check.js <PR>` — test coverage validation and rule/skill duplication invariant check. CI runs this automatically.
-  - **Acceptance check** (human judgment): `node .claude/skills/orchestrator/acceptance-check.js <PR>` via `run_process` — full Q1-Q9 interactive review. **Always required for production code changes.** Never skip this — even when the diff looks trivial. (Lesson: Sprint 2026-04-05c — skipping the full acceptance check caused a UI requirement to be missed on #599.)
+  - **Preflight check** (mechanical): `node .claude/skills/orchestrator/preflight-check.js <PR>` — test coverage validation, rule/skill duplication invariant check, and public-artifact language check. CI runs this automatically.
+  - **Acceptance check** (human judgment): `node .claude/skills/orchestrator/acceptance-check.js <PR>` via `run_process` — full Q1-Q11 interactive review. **Always required for production code changes.** Never skip this — even when the diff looks trivial. (Lesson: Sprint 2026-04-05c — skipping the full acceptance check caused a UI requirement to be missed on #599.)
+
```
[the remark this answers is not in the document]
Comment (ms2sato) [reply .claude/skills/orchestrator/core-responsibilities.md:143]: Addressed in commit c10bedc — updated Lines 92 and 113 (Trigger and Flow) from Q1-Q9 to Q1-Q11 to match the rest of the document.
